### ISSUE TITLE
LBFGS image quality

### ISSUE THREAD

**Issue by neale** [bethgelab/foolbox#53]
I'm wondering about the default parameters for the lbfgs attack. 
When I use the example code to attack an image with lbfgs, I get much worse results in terms of distance from adversarial to real, compared to the images given in [Exploring the space of adversarial images](https://arxiv.org/pdf/1510.05328.pdf)

I tried changing the initial C value within the optimization routine but it didn't help. Is the LBFGS attack supposed to make convincing adversarial images? I've also tried different methods of preprocessing, and also using a pytorch model. 

Here's is the image comparison [Images](http://imgur.com/a/FqpLN)
For the paper lbfgs implementation I'm using [this](https://github.com/tabacof/adversarial/tree/master/lbfgsb) repo mentioned in the paper. Its just not very flexible and uses torch7. 

Here's the test code I'm using for foolbox. 
```python
import ...
keras.backend.set_learning_phase(0)
kmodel = ResNet50(weights='imagenet')
preprocessing = (numpy.array([104, 116, 123]), 1)
fmodel = foolbox.models.KerasModel(kmodel, bounds=(0, 255), preprocessing=preprocessing)

image = scipy.misc.imread('bee.jpg')
image = scipy.misc.imresize(image, (224, 224))
label = numpy.argmax(fmodel.predictions(image))

# apply attack on source image
attack  = foolbox.attacks.LBFGSAttack(fmodel)
adversarial = attack(image[:,:,::-1], label)
```
I'm running the latest master

**Comment by wielandbrendel** [comment]
@neale thanks for the report! Did you check that it's not just channel flipping? The resnet expects channels to be BGR (not RGB as usual), that's why we are feeding `image[:,:,::-1]` to the attack. As a result, the adversarial you get should also invert channel ordering before using plotting them. Also, is the size of the perturbations really different? If yes, may I ask you to provide us with the images as numpy arrays (i.e. original image and the perturbations you get with foolbox lbfgs and the other lbfgs implementation)?

**Comment by neale** [comment]
I tried just changing it back to RGB, and I get a difference, though not what I think we're looking for. 
I just applied `adversarial = adversarial[:, :, ::-1]` before plotting. I get this [image](http://imgur.com/a/lM5Px)

I hope its something embarrassingly simple on my end. I know everyone (scipy, the other implementation) is using an old Fortran lib for LBFGS so I don't think anything is different there. 

I'll update this with the numpy arrays when I get them.

**Comment by neale** [comment]
I got it. There isn't any difference if you take the norm of the perturbations. The nice looking image from the paper implementation is a gradient map taken from different layers, not a per-pixel distance that you have. 
For future reference, the LBFGS adversarials look like the ones [we know and love](http://imgur.com/a/lM5Px) if you perform `adversarial[:,:,::-1]/255` 
Thanks a lot for the help,

**Comment by wielandbrendel** [comment]
I am glad you could solve this issue - thanks for digging! 👍
